packages/mwp-tracking-plugin/src/util/avro.js

```diff
@@ -194,6 +216,37 @@ const chapinEnvelopeDeserializer: Object => Deserializer = schema => {
 	};
 };
 
+const nextSerializer: Object => Serializer = schema => {
+	const analyticsSource = 'WEB';
+
+	return data => {
+		const { record, metadata } = JSON.parse(data);
+		const viewId = uuidv1();
+		const analytics = {
+			timestamp: Date.now(),
+			schemaFullname: `${schema.namespace}.${schema.name}`,
+			schemaVersion: schema.doc,
+			source: analyticsSource,
+			data: JSON.stringify(record),
+		};
+		const payload = {
+			records: [analytics],
+			viewId,
+			...metadata,
+		};
+		const encodingForNewPattern = JSON.stringify(payload);
```

Comment:
not a blocker but just noting a repeating lesson we don't often seen to learn. "new" is a relative term and leaving New in a name will be very confusing for future engineers that might find this if when new becomes old

Reply:
that's a very good point 😁